lib/runner/process-listener.js

```diff
@@ -49,10 +49,10 @@ module.exports = class {
 
   onExit(code) {
     if (code > 0) {
-      this.setExitCode(code);
+      this.process.exitCode = code;
+    } else {
+      this.process.exitCode = this.exitCode;
```

Comment:
this can be written more elegantly as `this.process.exitCode = code || this.exitCode`;